services/director-v2/src/simcore_service_director_v2/modules/docker_daemon.py

```diff
@@ -0,0 +1,51 @@
+""" Module responsible of interacting with docker daemon
+
+
+"""
+# TODO: code below simply copied from old director
+import logging
+
+import aiodocker
+from asyncio_extras import async_contextmanager  # Lt's try to avoid this??
+
+log = logging.getLogger(__name__)
+
+
+# Module's setup logic ---------------------------------------------
+
+
+# Module's business logic ---------------------------------------------
+
+
+@async_contextmanager
+async def docker_client() -> aiodocker.docker.Docker:
+    try:
+        client = aiodocker.Docker()
+        yield client
+    except aiodocker.exceptions.DockerError:
+        log.exception(msg="Unexpected error with docker client")
+        raise
+    finally:
+        await client.close()
```

Comment:
I always forget. If the code inside the context manger block raises a DockerError, will this trap it?

Reply:
yes , the decorator ``async_contextmanager`` takes care of that .. it will call ``next`` on the returned iterator and the exception will raise inisde